Hashtable.c

```diff
@@ -141,6 +205,10 @@ void Hashtable_clear(Hashtable* this) {
    assert(Hashtable_isConsistent(this));
 }
 
+static inline size_t inc_index(size_t index, size_t size) {
+   return ++index != size ? index : 0;
```

Comment:
Not sure if the compiler is forced by the standard to evaluate this in any particular order.

Reply:
This is enforced by the C standard (see [6.5.3.1 Prefix increment and decrement operator](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf)).

Reply:
More importantly there's the required sequencing in §6.5.15 paragraph 4. As well as §6.5.16.